From b19914aebaaa9235b42048b3fe2182057bf7968f Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 9 Jan 2023 16:59:03 -0700
Subject: [PATCH] Update chk.py

---
 molmod/io/chk.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/molmod/io/chk.py b/molmod/io/chk.py
index 5a04e03..1e55893 100644
--- a/molmod/io/chk.py
+++ b/molmod/io/chk.py
@@ -141,8 +141,8 @@ def dump_chk(filename, data):
                 if value.dtype.fields is not None:
                     raise TypeError('Arrays with fields are not supported.')
                 shape_str = ','.join(str(i) for i in value.shape)
-                if issubclass(value.dtype.type, (str, np.unicode, np.bytes_)):
-                    value = value.astype(np.unicode)
+                if issubclass(value.dtype.type, (str, np.unicode_, np.bytes_)):
+                    value = value.astype(str)
                     for cell in value.flat:
                         if len(cell) >= 22:
                             raise ValueError('In case of string arrays, a string may contain at most 21 characters.')
